code-challenges.js

```diff
@@ -8,62 +8,160 @@
 
 // Pseudo code:
 
+// -input is going to use two strings (2 of the following fruit1, fruit2, fruit3, fruit4)
+// -output is going to be whichever string has the most characters
+// -process begins by creating a function named mostFruitChar
+// -pass the 2 strings for comparison into the function string1, string2
+// -determine the characters in each of the strings using the .length built in method
+// -compare both strings to determine which has more characters using a comparison operator (>) and conditional statement if/else
+// -return string1 if more characters are present than string2 (from conditional statement if)
+// -return string2 if more characters are present than string1 (from conditional statement else)
+
 // Set one:
 const fruit1 = "apple"
 const fruit2 = "banana"
 // Expected outcome: "banana"
 
+const mostFruitChar = (string1, string2) =>{
+    if (string1.length > string2.length){
+        return string1
+    }
+    else {
+        return string2
+    }
+}
+console.log(mostFruitChar(fruit1, fruit2))
+       //outcome: banana
+
 // Set two:
 const fruit3 = "cherry"
 const fruit4 = "kiwi"
 // Expected outcome: "cherry"
 
+console.log(mostFruitChar(fruit3, fruit4))
+       //outcome: cherry
+
+
 // --------------------1) Create a function that takes in a number and determines if the given number is below boiling point, at boiling point, or above boiling point. Boiling point is 212 degrees Fahrenheit. Use the test variables provided below.
 
 // Pseudo code:
+// -input is 1 number from a possible 3 variables (temperature1, temperature2, temperature3)
+// -output is if the test variable is below, at, or above boiling point of 212 degrees
+// -process is to create a function named boilingPoint
+// -pass the parameter of temp into the function for comparison to boiling point. **** learned by accident parameter does not need parentheses (at least in this instance)
+// -determine if variable is below, at or above boiling point using comparison operator (< or >) and boiling point of 212  degrees and conditionals if/else
+// -return (temperature1) 42 is below boiling point if it is below 212 using string interpolation
+// -return (temperature2) 350 is above boiling point if it is above 212 using string interpolation
+// -return (temperature3) 212 is at boiling point if it is equal to 212 using string interpolation
+
+const boilingPoint = temp => {
+    if (temp < 212){
+        return `${temperature1} is below boiling point`;
+    }
+    else if (temp > 212){
+        return `${temperature2} is above boiling point`;
+    }
+    else  {
+        return `${temperature3} is at boiling point`;
+    }
+}
```

Comment:
Remember, the parameter is a placeholder for the argument, so you don't want to be putting the actual given variables inside your function, because then the function is not dynamic. For example, what if you didn't know the values of `temperature1`, `temperature2` and `temperature3`? The parameter (in this case `temp`) becomes the argument (whatever you pass into your function when you invoke the function).